### PR TITLE
main/linux-vanilla: Add missing build dependencies

### PR DESCRIPTION
**Pull request by deedubs**
Attempted to get kernel building on CircleCI has highlighted these missing deps

flex and bison were causing the configure step to fail

openssl-dev is also required during the build

```
/home/alpine/project/testing/linux-vanilla/src/linux-4.17/scripts/extract-cert.c:21:25: fatal error: openssl/bio.h: No such file or directory
 #include <openssl/bio.h>
```

### CODE REVIEW

File: main/linux-vanilla/APKBUILD [hunk 1]
```diff
@@ -12,7 +12,7 @@ pkgdesc="Linux vanilla kernel"
 url="http://kernel.org"
 depends="mkinitfs"
 _depends_dev="perl gmp-dev elfutils-dev bash"
-makedepends="$_depends_dev sed installkernel bc linux-headers linux-firmware"
+makedepends="$_depends_dev flex bison sed installkernel bc linux-headers linux-firmware openssl-dev"
```
Comment (ncopa) [inline main/linux-vanilla/APKBUILD]: we currently use libressl, so if this is really needed, then it should be `libressl-dev`.

I wonder why we don't hit this issue on our builders?

**Comment by deedubs** [comment]
Yeah I was chatting about this with _ikke_ on IRC. The assumption was that the build machines aren't as minimal as originally thought. 

I'll update to libressl-dev.

**Comment by deedubs** [comment]
@ncopa Updated to `libressl-dev`

**Comment by ncopa** [comment]
i still don't understand why this is needed. I can not reproduce the problem you are trying to solve.

**Comment by deedubs** [comment]
Sorry, I forgot you were missing some context from IRC.  The builds are failing from a 'clean' build env with missing deps https://circleci.com/gh/massivekube/aports/55.

**Comment by ncopa** [comment]
aha, its for 4.17 kernel.

> linux-vanilla: Applying patch-4.17.8.xz

I suppose we don't need those til we upgrade the kernel.

**Comment by ncopa** [comment]
this should be fixed with d8adf28a6ff7a7527a2d078f5800067b2f506634

**Pull request closed without merging**